### PR TITLE
[bugfix] dynamic property render in snippets

### PR DESCRIPTION
**Pull request by ajaystha42**
Addresses: https://github.com/restarone/violet_rails/issues/1355
Demo: 

https://user-images.githubusercontent.com/55496123/220007755-4d4fa5f6-0534-4c7a-a499-06978793219c.mov

### CODE REVIEW

**Comment by donrestarone** [comment]
@ajaystha42 did you test your code? The generated template renders nothing because it scopes by all attributes:

Screen Shot 2023-01-17 at 3 06 36 PM
Screen Shot 2023-01-17 at 3 05 53 PM

**Comment by ajaystha42** [comment]
> @ajaystha42 did you test your code? The generated template renders nothing because it scopes by all attributes:
> 
> Screen Shot 2023-01-17 at 3 06 36 PM
> 
> Screen Shot 2023-01-17 at 3 05 53 PM

@donrestarone , it looks like the scoping fails for an array only. I will fix it!

**Comment by donrestarone** [comment]
review this PR in the CMS: https://github.com/restarone/comfortable-mexican-sofa/pull/12

File: Gemfile [hunk 1]
```diff
@@ -116,4 +116,4 @@ gem "redis-namespace", "~> 1.8"
 
 gem 'stripe-rails'
 
-gem 'devise-two-factor', "4.0.2"
+gem 'devise-two-factor', "4.0.2"
```
Comment (donrestarone) [inline Gemfile:119]: please remove this change

File: Gemfile.lock [hunk 2]
```diff
@@ -577,4 +577,4 @@ RUBY VERSION
    ruby 3.0.0p0
 
 BUNDLED WITH
-   2.1.4
+   2.1.4
```
Comment (donrestarone) [inline Gemfile.lock:580]: please remove this change

**Comment by donrestarone** [comment]
@ajaystha42 could you add a test to verify that the correct snippet is rendered?

**Comment by ajaystha42** [comment]
> @ajaystha42 could you add a test to verify that the correct snippet is rendered?

@donrestarone I have added a test for this: https://github.com/restarone/violet_rails/pull/1364/files#diff-bfe11616aee067548a7858d9cbe05d4040a61f437e0576be4a64b699ebff8aa7R786

File: app/helpers/content_helper.rb [hunk 3]
```diff
@@ -52,7 +51,7 @@ def render_api_namespace_resource(api_namespace_slug, options = {})
     
     api_namespace = ApiNamespace.find_by(slug: api_namespace_slug)
     api_resources = api_namespace.api_resources
-
+
```
Comment (donrestarone) [inline app/helpers/content_helper.rb:54]: @ajaystha42 please remove the blank line changes.

File: app/controllers/comfy/admin/api_namespaces_controller.rb [hunk 4]
```diff
@@ -39,9 +39,15 @@ def show
     params[:q] ||= {}
     @api_resources_q = @api_namespace.api_resources.ransack(params[:q])
     @api_resources = @api_resources_q.result.paginate(page: params[:page], per_page: 10)
-
```
Comment (donrestarone) [inline app/controllers/comfy/admin/api_namespaces_controller.rb:42]: @ajaystha42 please remove the blank line changes.

**Comment by donrestarone** [comment]
@ajaystha42 after you address my blank line changes (please note moving forward changes are to be terse) change the Branch we use for restarone/comfortable-mexican-sofa in the Gemfile to use the branch ajaystha42feat/parser_modifications for further testing

File: Gemfile [hunk 5]
```diff
@@ -31,7 +31,7 @@ gem 'ros-apartment', require: 'apartment'
 gem 'ros-apartment-sidekiq', require: 'apartment-sidekiq'
 gem 'apartment-activejob'
 gem 'devise'
-gem "comfortable_mexican_sofa",git: 'https://github.com/restarone/comfortable-mexican-sofa', tag: '3.4'
+gem "comfortable_mexican_sofa",git: 'https://github.com/ajaystha42/comfortable-mexican-sofa', branch: 'feat/parser_modifications'
```
Comment (donrestarone) [inline Gemfile:34]: @ajaystha42 this should be restarone/cms

**Comment by donrestarone** [comment]
this task is blocked by this: https://github.com/restarone/violet_rails/issues/1393

we have to upgrade to a non-vulnerable version of loofah in the CMS before we can re-bundle. Because when we change the bundle we trip this brakeman check

File: Gemfile (hunk 5 above)
Comment (ajaystha42) [reply Gemfile:34]: @donrestarone we made the changes in cms for this task, 
To test the changes in review app, we need to make this change before merging the changes in https://github.com/restarone/comfortable-mexican-sofa/pull/13
This change will be reverted once the UAT passes

Comment (donrestarone) [reply Gemfile:34]: @ajaystha42 you need to point at restarone/cms., as I said before. We aren't deploying your fork, only our own

Comment (ajaystha42) [reply Gemfile]: @donrestarone  I pointed it to restarone/cms. But, the changes wont work until we merge [this](https://github.com/restarone/comfortable-mexican-sofa/pull/13)

**Comment by donrestarone** [comment]
copy pasting snippet is still not working, we need to verify that the generated snippet works out of the box

Screen Shot 2023-02-16 at 10 15 39 PM
Screen Shot 2023-02-16 at 10 15 21 PM


@Ayon95 could you take a look? This issue is affecting sipshucksip.com

**Comment by Ayon95** [comment]
@donrestarone  The index snippet is working now. The snippet that was being used was not right. Can you create some resources for `test` API namespace and check if they're being rendered on the page? The snippet content is showing up.

snippet

snippet-content


tester-page

**Comment by ajaystha42** [comment]
@donrestarone  Please review this [PR](https://github.com/restarone/comfortable-mexican-sofa/pull/14). I will point to the cms/restarone once you merge it in the main repo from the forked repo. Also, here is the video to demonstrate the working of dynamic snippets: 

https://user-images.githubusercontent.com/55496123/220007638-1a0d87f0-8f68-47b9-8e6d-a7a6ba242680.mov

**Comment by donrestarone** [comment]
@ajaystha42 there are 2 PR's of yours that do the same thing on https://github.com/restarone/comfortable-mexican-sofa/pulls

this PR is using the branch here: https://github.com/restarone/comfortable-mexican-sofa/pull/13 and as I have mentioned before, the default generated snippet is not working. 

Lets revisit this with fresh eyes later

**Comment by ajaystha42** [comment]
@donrestarone  The snippets are working now: Please have a look here: https://review-1364.violet-test.net/test-page
Screenshot 2023-02-22 at 09 34 53
Screenshot 2023-02-22 at 09 34 36
Screenshot 2023-02-22 at 09 34 21

**Comment by donrestarone** [comment]
LGTM on the review app, deploying RC ✅ 

After release candidate is verified, we can merge https://github.com/restarone/comfortable-mexican-sofa/pull/16, bump the version on the CMS and change to the latest version before merging to master

**Pull request merged**